File: frontend/src/api/diary.ts
/* today_book/frontend/src/api/diary.ts */
import apiClient from './client'

export interface Diary {
  id: number
  user_id: number
  date: string
  title?: string | null
  content?: string | null
  mood?: string | null
  weather?: string | null
  tags?: string[] | null
  created_at: string
  updated_at: string
}

export interface CalendarDay {
  date: string
  has_entry: boolean
  mood?: string
}

export const diaryApi = {
  list: (params?: { year?: number; month?: number; page?: number; page_size?: number }) =>
    apiClient.get<Diary[]>('/diary/', { params }),

  getByDate: (date: string) =>
    apiClient.get<Diary>(`/diary/${date}`),

  upsert: (data: Partial<Diary>) =>
    apiClient.post<Diary>('/diary/', data),

  delete: (date: string) =>
    apiClient.delete(`/diary/${date}`),

  getCalendar: (year: number, month: number) =>
    apiClient.get<CalendarDay[]>(`/diary/calendar/${year}/${month}`),
}

File: backend/app/schemas/diary.py
# today_book/backend/app/schemas/diary.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class DiaryCreate(BaseModel):
    date: str  # YYYY-MM-DD
    title: Optional[str] = None
    content: Optional[str] = None
    mood: Optional[str] = None
    weather: Optional[str] = None
    tags: Optional[list[str]] = None


class DiaryUpdate(BaseModel):
    title: Optional[str] = None
    content: Optional[str] = None
    mood: Optional[str] = None
    weather: Optional[str] = None
    tags: Optional[list[str]] = None


class DiaryResponse(BaseModel):
    id: int
    user_id: int
    date: str
    title: Optional[str]
    content: Optional[str]
    mood: Optional[str]
    weather: Optional[str]
    tags: Optional[list[str]]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class DiaryCalendarDay(BaseModel):
    date: str
    has_entry: bool
    mood: Optional[str] = None
